metrics-collectd/src/test/java/com/codahale/metrics/collectd/Receiver.java

```diff
@@ -0,0 +1,81 @@
+package com.codahale.metrics.collectd;
+
+import org.collectd.api.Notification;
+import org.collectd.api.ValueList;
+import org.collectd.protocol.Dispatcher;
+import org.collectd.protocol.UdpReceiver;
+import org.junit.rules.ExternalResource;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+public final class Receiver extends ExternalResource implements Runnable {
+
+    private int port;
+    private UdpReceiver receiver;
+    private Thread receiverThread;
+    private BlockingQueue<ValueList> queue;
+
+    public Receiver(int port) {
+        this.port = port;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        queue = new LinkedBlockingQueue<ValueList>();
+        receiver = new UdpReceiver(new QueueDispatcher(queue));
```

Comment:
We can avoid creating the `QueueDispatcher` class and just create a new anonymous `Dispatcher` class, because it's used only in one place.
```java
       new Dispatcher() {
            @Override
            public void dispatch(ValueList values) {
                queue.offer(new ValueList(values));
            }

            @Override
            public void dispatch(Notification notification) {
                throw new UnsupportedOperationException();
            }
        }
```